From e6f0a243c346c91b5510297106a87547c1aca862 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 03 Jun 2021 12:10:38 +0300
Subject: [PATCH] [SW-34388] gaudi2: add code to set the uart pads

This patch sets the pad registers for UART RX and TX.

Change-Id: I3da7a9b4a6a1e075ead3fc1bd0baef22ced9c76f
---

diff --git a/gaudi2/onix.c b/gaudi2/onix.c
index 34a2e6b..4c28fb1 100644
--- a/gaudi2/onix.c
+++ b/gaudi2/onix.c
@@ -11,6 +11,7 @@
 #include <pll.h>
 #include <uart.h>
 #include <cmd_module.h>
+#include <psoc_global_conf_regs.h>
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
@@ -26,6 +27,8 @@
 
 void hl_uart_init(void)
 {
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
 	preboot_uart_init(SYS_CLOCK_FREQUENCY / (HL_UART_BAUDRATE / 2));
 }
 
diff --git a/gaudi2/pldm.c b/gaudi2/pldm.c
index 11b882f..dc27f54 100644
--- a/gaudi2/pldm.c
+++ b/gaudi2/pldm.c
@@ -10,6 +10,7 @@
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
+#include <psoc_global_conf_regs.h>
 
 #define SYS_CLOCK_FREQUENCY	155000ULL
 #define UART_CLOCK_FREQUENCY	11900ULL
@@ -27,6 +28,8 @@
 
 void hl_uart_init(void)
 {
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
 }
 
